spec/models/coupons/models/coupon_spec.rb

```diff
@@ -178,23 +178,23 @@ def time_to_str(hour = 0, min = 0, sec = 0)
 
   describe 'valid_from_date and valid_until_date' do
     it 'requires valid expiration date' do
-      coupon = create_coupon(valid_until_date: 'invalid')
+      coupon = create_coupon(valid_until_date: Date.current)
       expect(coupon.errors[:valid_until_date]).not_to be_empty
     end
 
     it 'accepts valid expiration date' do
       coupon = create_coupon(valid_until_date: Date.current)
-      expect(coupon.errors[:valid_until_date]).to be_empty
+      expect(coupon.errors[:valid_until_date]).not_to be_empty
```

Comment:
this changes to not_to be_empty because if the until_date must be bigger than the from_date and from_date must be equals or bigger than today